flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/PipelinedSubpartitionView.java

```diff
@@ -39,13 +39,15 @@
 	private final AtomicBoolean isReleased;
 
 	PipelinedSubpartitionView(PipelinedSubpartition parent, BufferAvailabilityListener listener) {
+		super(parent);
+
 		this.parent = checkNotNull(parent);
 		this.availabilityListener = checkNotNull(listener);
 		this.isReleased = new AtomicBoolean();
 	}
 
 	@Override
-	public Buffer getNextBuffer() {
+	protected Buffer getNextBufferInternal() {
```

Comment:
Actually, a lot of the methods along these calls should probably be marked `@Nullable`. Since you touched the `getNextBufferInternal()`, can you at least mark this (and maybe some calls along the call stack if I say pretty please?). You can do so in a separate `[hotfix]` commit to keep this separate

Reply:
I will add the hotfix commit for it.